app/models/invoice.rb

```diff
@@ -17,22 +17,21 @@ def self.unshipped_items
     .order('invoices.created_at asc')
   end
 
-  def self.highest_revenue_date(item_id)
-    invoices = joins(:invoice_items)
-    .joins('right join transactions on transactions.invoice_id=invoices.id')
-    .select(invoices: :updated_at)
-    .where(invoice_items: {item_id: item_id}, transactions: {result: 1}, invoices: {status: 1})
-    .order(Arel.sql('sum(invoice_items.quantity * invoice_items.unit_price) desc, invoices.updated_at desc'))
-    .group(:updated_at)
-    .limit(1)
-    .pluck(:updated_at)
-  end
-
   def item_sale_price
-    self.items.joins(:invoice_items).select('items.*, invoice_items.unit_price as sale_price, invoice_items.quantity as sale_quantity')
+    items
+    .joins(:invoice_items)
+    .select('items.*, invoice_items.unit_price as sale_price, invoice_items.quantity as sale_quantity')
```

Comment:
@mkrumholz This is probably a question for the view, but did you needed to create all of these queries for the total revenue?